server/test/performance/performance_test_annotations_backend.py

```diff
@@ -0,0 +1,215 @@
+            }
+            df = pd.DataFrame(annon_dict)
+            df_size = sys.getsizeof(df) / 1024 ** 2
+            self.test_notes[dataset_name][f"num_categories_{num_cat}"][f"max_label_{max_labels}"]["data_frame"] = {
+                "creation_time": str(elapsed()),
+                "size": f"{df_size} mb",
+            }
+            try:
+                matrix = encode_matrix_fbs(matrix=df, row_idx=None, col_idx=df.columns)
+                matrix_size = sys.getsizeof(matrix) / 1024 ** 2
+                self.test_notes[dataset_name][f"num_categories_{num_cat}"][f"max_label_{max_labels}"]["fbs_matrix"] = {
+                    "creation_time": str(elapsed()),
+                    "size": f"{matrix_size} mb",
+                }
+                return matrix
+            except Exception as e:
+                print(f"Issue creating fbs matrix: {e}, for {dataset_name}")
+                return []
+
+    def send_put_request(self, dataset_url, data):
+        url = self.url_base + f"{dataset_url}/api/v0.2/annotations/obs"
+        with self.elapsed_timer() as elapsed:
+            try:
+                headers = {"Content-Type": "application/octet-stream", "Cookie": cookie}
+                response = requests.put(url=url, data=data, headers=headers)
+            except Exception as e:
+                print(f"Issue with put request: {e}")
+                return None, elapsed()
+            return response, elapsed()
+
+    def test_categories_max_label_matrix(self, dataset_name):
+        for unique_labels in self.max_labels:
+            for category_count in self.annotations_category_count:
+                print(f"Starting dataset: {dataset_name}, categories: {category_count}, labels: {unique_labels}")
+                fbs_matrix = self.create_matrix(dataset_name, category_count, unique_labels)
+                if self.test_datasets[dataset_name]["dataset_url"] and fbs_matrix:
+                    response, response_time = self.send_put_request(
+                        self.test_datasets[dataset_name]["dataset_url"], fbs_matrix
+                    )
+                    if response is None:
+                        self.test_notes[dataset_name][f"num_categories_{category_count}"][f"max_label_{unique_labels}"][
+                            "put_request"
+                        ] = {"response_status": "failed", "request_time": str(response_time)}
+                    else:
+                        self.test_notes[dataset_name][f"num_categories_{category_count}"][f"max_label_{unique_labels}"][
+                            "put_request"
+                        ] = {"response_status": response.status_code, "request_time": str(response_time)}
+
+
+def test_all_datasets():
+    """
+    Run time is dependent on number of datasets, dataset size, number of categories/number being tested and number of
+     unique label counts being tested. However it generally takes a long time. I recommend running this in tmux
```

Comment:
Can you put a number to it. Is a long time 1 hour, 30min, 10min, ect.... It will help to know what a long time roughly means.

Reply:
its completely dependent on the number of datasets the size of the datasets etc it can take hours. This is just a suggestion that people use tmux if they are testing lots of datasets/larger datasets/a lot of combos of max labels and num of categories

Reply:
I don't have a better suggestion of phrasing, but "a long time" is not particularly helpful. Maybe remove it all together.